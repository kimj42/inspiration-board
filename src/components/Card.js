import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  render() {
    return (
      <div className="card">
        {this.props.text} {this.props.emoji}
      </div>
    )
  }
}

Card.propTypes = {

};

export default Card;
