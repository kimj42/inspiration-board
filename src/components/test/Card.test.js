import React from 'react';
import Card from '../Card.js';
import { shallow } from 'enzyme';

describe('Card', () => {

  test('card must work with test and emoji passed in as string', () => {
    const wrapper = shallow( <Card text={'hello testing card'} emoji={'heart_eyes'} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('card does not break if test and emjoi does not show up', () => {
    const wrapper = shallow( <Card />);

    expect(wrapper).toMatchSnapshot();
  });

});
