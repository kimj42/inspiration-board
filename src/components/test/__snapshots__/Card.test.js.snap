// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card card does not break if test and emjoi does not show up 1`] = `
<div
  className="card"
>
   
</div>
`;

exports[`Card card must work with test and emoji passed in as string 1`] = `
<div
  className="card"
>
  hello testing card
   
  heart_eyes
</div>
`;
